command/agent/config_parse.go

```diff
@@ -141,6 +141,7 @@ func extraKeys(c *Config) error {
 	// Remove HostVolume extra keys
 	for _, hv := range c.Client.HostVolumes {
 		removeEqualFold(&c.Client.ExtraKeysHCL, hv.Name)
+		removeEqualFold(&c.Client.ExtraKeysHCL, "host_volume")
```

Comment:
Yeah, the HCL parse tree keeps duplicate entries for anything that's a slice, and the extra keys bubble up. I don't think it's possible to remove the extra keys unambiguously in the parser.

Reply:
huh thanks!